packages/handlersjs-core/lib/daemons/scheduler.spec.ts

```diff
@@ -0,0 +1,104 @@
+import { Scheduler } from './scheduler';
+
+describe('Scheduler', () => {
+
+  let scheduler: Scheduler;
+  const task = jest.fn();
+
+  // checks if the scheduler is running
+  const isRunning = async () => {
+
+    task.mockClear();
+    await new Promise((r) => setTimeout(r, 20)); // sleep 20ms
+    const callAmount = task.mock.calls.length;
+
+    return callAmount >= 3 && callAmount <= 4;
+
+  };
```

Comment:
This is too elaborate. Testing whether the task has been run can be done with Jest's functions (e.g. `expect(task).toHaveBeenCalledTimes(0)`, or even better `expect(task).not.toHaveBeenCalled()`); testing whether the interval is reset can be done by mocking `setInterval` and checking whether it has been called.

Reply:
I split this up in 2 functions, one for asserting the scheduler is running and one for asserting that it is not running

I couldn't solve the main issue though, because I can't assert an exact amount of calls to be expected since in this case, when waiting 20ms, it either might be called 3 times or 4 times and both of them are correct behavior

Reply:
would changing it to 1000ms give us a certain 4? ( Not sure if this is still an issue )

Reply:
there was a bigger issue at play which caused the result of expectIsRunning to be neglected actually, whether it was true/false, but I fixed it now and now it's always called 4 times